js/gestionPresupuesto.js

```diff
@@ -1,25 +1,110 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
 
 // TODO: Variable global
+let presupuesto = 0;
+let gastos = [];
+let idGasto = 0;
 
 
-function actualizarPresupuesto() {
-    // TODO
+function actualizarPresupuesto(nuevoPresupuesto) {
+    return (nuevoPresupuesto >= 0)?(presupuesto = nuevoPresupuesto, nuevoPresupuesto):(console.log("Error, numero negativo"), -1);
 }
 
+
 function mostrarPresupuesto() {
-    // TODO
+    return `Tu presupuesto actual es de ${presupuesto} €`;
+}
+
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+    
+    this.descripcion=descripcion;
+    (valor>0 && !valor.isNaN)?(this.valor = valor):(this.valor = 0);
+    (!isNaN(Date.parse(fecha)))?(this.fecha = Date.parse(fecha)):(this.fecha = Date.now());
+    (etiquetas)?(this.etiquetas = etiquetas):(this.etiquetas = []);
```

Comment:
Aquí, en lugar de asignar directamente los datos que te pasan, quizá puedes llamar a la función `this.anyadirEtiquetas`, con el objetivo de eliminar posibles duplicados.